src/ApiCore/AuthWrapper.php

```diff
@@ -55,6 +63,63 @@ public function __construct(FetchAuthTokenInterface $fetchAuthTokenInterface, ca
         $this->authHttpHandler = $authHttpHandler;
     }
 
+    /**
+     * @param array $args {
+     *     An array of required and optional arguments.
+     *
+     *     @type CredentialsLoader $credentialsLoader
+     *           A CredentialsLoader object created using the Google\Auth library.
+     *     @type array $scopes
+     *           A string array of scopes to use when acquiring credentials.
+     *     @type callable $authHttpHandler A handler used to deliver PSR-7
+     *           requests specifically for authentication. Should match a
+     *           signature of
+     *           `function (RequestInterface $request, array $options) : ResponseInterface`.
+     *     @type bool $enableCaching
+     *           Optional. Enable caching of access tokens. Defaults to true.
+     *     @type CacheItemPoolInterface $authCache
+     *           Optional. A cache for storing access tokens. Defaults to a simple in memory implementation.
+     *     @type array $authCacheOptions
+     *           Optional. Cache configuration options.
+     * }
+     * @return AuthWrapper
+     * @throws \Exception
```

Comment:
I believe this should be a `ValidationException`. Also, `ApplicationDefaultCredentials::getCredentials()` throws a `\DomainException`. We may want to catch that and bubble up a `ValidationException` for consistency, WDYT?

Reply:
I have converted to `ValidationException`, PTAL.